TestDifficulty fails with fresh genesis blocks

To reproduce:
1. Generate new genesis blocks with fresh timestamps.
2. Run `go test ./... -run=TestDifficulty`.

Stack trace:
```
--- FAIL: TestDifficulty (6.95s)
    for_all_nets.go:22: Running test for kaspa-mainnet
    difficultymanager_test.go:54: ValidateAndInsertBlock: ErrBlockIsTooMuchInTheFuture
        The block timestamp is in the future.
        github.com/kaspanet/kaspad/domain/consensus/processes/blockvalidator.(*blockValidator).checkBlockTimestampInIsolation
                ./kaspad/domain/consensus/processes/blockvalidator/block_header_in_isolation.go:51
        github.com/kaspanet/kaspad/domain/consensus/processes/blockvalidator.(*blockValidator).ValidateHeaderInIsolation
                ./kaspad/domain/consensus/processes/blockvalidator/block_header_in_isolation.go:19
        github.com/kaspanet/kaspad/domain/consensus/processes/blockprocessor.(*blockProcessor).validatePreProofOfWork
                ./kaspad/domain/consensus/processes/blockprocessor/validateandinsertblock.go:294
        github.com/kaspanet/kaspad/domain/consensus/processes/blockprocessor.(*blockProcessor).validateBlock
                ./kaspad/domain/consensus/processes/blockprocessor/validateandinsertblock.go:254
        github.com/kaspanet/kaspad/domain/consensus/processes/blockprocessor.(*blockProcessor).validateAndInsertBlock
                ./kaspad/domain/consensus/processes/blockprocessor/validateandinsertblock.go:37
        github.com/kaspanet/kaspad/domain/consensus/processes/blockprocessor.(*blockProcessor).ValidateAndInsertBlock
                ./kaspad/domain/consensus/processes/blockprocessor/blockprocessor.go:124
        github.com/kaspanet/kaspad/domain/consensus.(*consensus).ValidateAndInsertBlock
                ./kaspad/domain/consensus/consensus.go:64
        github.com/kaspanet/kaspad/domain/consensus/processes/difficultymanager_test.TestDifficulty.func1.1
                ./kaspad/domain/consensus/processes/difficultymanager/difficultymanager_test.go:52
        github.com/kaspanet/kaspad/domain/consensus/processes/difficultymanager_test.TestDifficulty.func1
                ./kaspad/domain/consensus/processes/difficultymanager/difficultymanager_test.go:155
        github.com/kaspanet/kaspad/domain/consensus/utils/testutils.ForAllNets
                ./kaspad/domain/consensus/utils/testutils/for_all_nets.go:23
        github.com/kaspanet/kaspad/domain/consensus/processes/difficultymanager_test.TestDifficulty
                ./kaspad/domain/consensus/processes/difficultymanager/difficultymanager_test.go:17
        testing.tRunner
                /usr/lib/go/src/testing/testing.go:1123
        runtime.goexit
                /usr/lib/go/src/runtime/asm_amd64.s:1374
FAIL
exit status 1

```

--- a/domain/consensus/processes/blockvalidator/block_header_in_isolation.go
+++ b/domain/consensus/processes/blockvalidator/block_header_in_isolation.go
@@ -47,7 +47,6 @@ func (v *blockValidator) checkParentsLimit(header *externalapi.DomainBlockHeader
 }
 
 func (v *blockValidator) checkBlockTimestampInIsolation(header *externalapi.DomainBlockHeader) error {
-
 	blockTimestamp := header.TimeInMilliseconds
 	now := mstime.Now().UnixMilliseconds()
 	maxCurrentTime := now + int64(v.timestampDeviationTolerance)*v.targetTimePerBlock.Milliseconds()

--- a/domain/consensus/processes/difficultymanager/difficultymanager_test.go
+++ b/domain/consensus/processes/difficultymanager/difficultymanager_test.go
@@ -2,6 +2,9 @@ package difficultymanager_test
 
 import (
 	"testing"
+	"time"
+
+	"github.com/kaspanet/kaspad/util/mstime"
 
 	"github.com/kaspanet/kaspad/domain/consensus/utils/consensushashing"
 
@@ -18,6 +21,14 @@ func TestDifficulty(t *testing.T) {
 		if params.DisableDifficultyAdjustment {
 			return
 		}
+		// This test generates 3066 blocks above genesis with at least 1 second between each block, amounting to
+		// a bit less then an hour of timestamps.
+		// To prevent rejected blocks due to timestamps in the future, the following safeguard makes sure
+		// the genesis block is at least 1 hour in the past.
+		if params.GenesisBlock.Header.TimeInMilliseconds > mstime.ToMSTime(time.Now().Add(-time.Hour)).UnixMilliseconds() {
+			t.Fatalf("TestDifficulty requires the GenesisBlock to be at least 1 hour old to pass")
+		}
+
 		params.K = 1
 		params.DifficultyAdjustmentWindowSize = 264
 

